src/main/scala/org/globalforestwatch/summarystats/gladalerts/GladAlertsRDD.scala

```diff
@@ -1,36 +1,42 @@
 package org.globalforestwatch.summarystats.gladalerts
 
 import cats.implicits._
-import geotrellis.contrib.polygonal._
+import geotrellis.layer.{LayoutDefinition, SpatialKey}
+import geotrellis.raster.summary.polygonal._
+import geotrellis.raster.summary.GridVisitor
 import geotrellis.raster._
 import geotrellis.raster.rasterize.Rasterizer
 import geotrellis.vector._
 import org.globalforestwatch.summarystats.SummaryRDD
+import org.globalforestwatch.util.Util.getAnyMapValue
+import org.globalforestwatch.util.{Geodesy, Mercantile}
+
+import scala.annotation.tailrec
 
 object GladAlertsRDD extends SummaryRDD {
 
   type SOURCES = GladAlertsGridSources
   type SUMMARY = GladAlertsSummary
   type TILE = GladAlertsTile
 
-  def getSources(window: Extent, kwargs: Map[String, Any]): Either[Throwable, SOURCES] = {
+  def getSources(key: SpatialKey, windowLayout: LayoutDefinition, kwargs: Map[String, Any]): Either[Throwable, SOURCES] = {
     Either.catchNonFatal {
-      GladAlertsGrid.getRasterSource(window, kwargs)
+      GladAlertsGrid.getRasterSource(key, windowLayout, kwargs)
     }
   }
 
-  def readWindow(rs: SOURCES, window: Extent): Either[Throwable, Raster[TILE]] =
-    rs.readWindow(window)
+  def readWindow(rs: SOURCES, windowKey: SpatialKey, windowLayout: LayoutDefinition): Either[Throwable, Raster[TILE]] =
+    rs.readWindow(windowKey, windowLayout)
 
   def runPolygonalSummary(raster: Raster[TILE],
                           geometry: Geometry,
                           options: Rasterizer.Options,
-                          kwargs: Map[String, Any]): SUMMARY = {
+                          kwargs: Map[String, Any]): PolygonalSummaryResult[SUMMARY] = {
+
     raster.polygonalSummary(
-      geometry = geometry,
-      emptyResult = new GladAlertsSummary(kwargs = kwargs),
+      geometry,
+      GladAlertsSummary.getGridVisitor(kwargs),
```

Comment:
Why do you use a different pattern here compared to other packages?
I would prefer if signature is as consistent as possible throughout all clusters.

I didn't use the kwargs in some of the other packages, but they are handy to have in case we need them at a later point.

Reply:
Ah ok, yeah I specifically had to do it this way to support passing kwargs, I can change all of them to do it this way.